src/Apps/Order/Components/__tests__/AddressVerificationFlow.jest.tsx

```diff
@@ -0,0 +1,38 @@
+import { screen } from "@testing-library/react"
+import { AddressVerificationFlowFragmentContainer } from "Apps/Order/Components/AddressVerificationFlow"
+import { setupTestWrapperTL } from "DevTools/setupTestWrapper"
+import { graphql } from "react-relay"
+
+jest.unmock("react-relay")
+
+const { renderWithRelay } = setupTestWrapperTL({
+  Component: AddressVerificationFlowFragmentContainer,
+  query: graphql`
+    query AddressVerificationFlow_Test_Query($address: AddressInput!)
+      @relay_test_operation {
+      verifyAddress(address: $address) {
+        ...AddressVerificationFlow_verifyAddress
+      }
+    }
+  `,
+})
+
+describe("AddressVerificationFlow", () => {
+  it(`renders the "Check your delivery address" variant when verificaiton is "NOT_FOUND"`, async () => {
+    renderWithRelay({
+      VerifyAddressType: () => ({
+        verificationStatus: "NOT_FOUND",
+      }),
+    })
+
+    await screen.findByText("Check your delivery address")
+
+    expect(
+      screen.getByText(
+        "The address you entered may be incorrect or incomplete. Please check it and make any changes necessary."
```

Comment:
**question [non-blocking]:** 

One more thing I noticed, are you expecting all elements to be present in the DOM immediately after rendering? If so, it would be consistent to use getByText() for all of them. However, if some elements might not be immediately available in the DOM, using findByText() for those would be more appropriate in that case.

*format via [Conventional Comments](https://conventionalcomments.org/)*

Reply:
You're right - the text that I'm expecting here is not available on the initial render. They are added in the `useEffect` hook that fires after that initial render. I have added an `await screen.findByText(...)` for these tests to wait for that hook to fire before making these assertions.